# Variables - thinking like a computer (scientist)

*Note*: this page is inspired by and partly adapted from [Brett and co-authors (2023)](https://odsti.github.io/cfd-textbook/variables_intro.html).

Imagine a person called Aaron. Aaron is studying in the United Kingdom. He wants to make a budget, so that he does not spend too much money each month.

His income every month is £1,100. He wants to know how much money he has left after he pays:

- His rent, which costs £575 every month.

- His bills, which cost £150 every month.

This is easy to calculate in Python:

In [1]:
1100 - 575 - 150

375

So he has £375 left after he pays his rent and bills.

But what if his landlord puts up his rent to £600 each month?

He could type all of the code again:

In [2]:
1100 - 600 - 150

350

But, this is not convenient. If we have more than three numbers, it will also get confusing.

Aaron needs *variables*.

## Assigning variables

Look at this code:

In [3]:
income = 1100

rent = 575

bills = 150

income - rent - bills

375

The first three lines...

```
income = 1100

rent = 575

bills = 150

```

...are called *variable assignment statements*. When we use an assignment statement, we make a new *variable*.

Look at the first line:

```
income = 1100
```

![](https://raw.githubusercontent.com/pxr687/Intro_Comp_Programming/refs/heads/main/images/variable_assignment_first.png)

This *variable assignment statement* has two *sides*:

- The *left* of the `=`

- The *right* of the `=`

![](https://raw.githubusercontent.com/pxr687/Intro_Comp_Programming/refs/heads/main/images/variable_assignment_left_right.png)

In this line of code, on the left of the `=` it says `income`. This is the *name* of the variable.

On the right of the `=` it says `1100`. This is the *value* of the variable.

![](https://raw.githubusercontent.com/pxr687/Intro_Comp_Programming/refs/heads/main/images/variable_assignment_name_val.png)

Creating a variable means our computer stores the *value* in memory. We can use the *name* to get the value:

In [4]:
# Show the value of the `income` variable.
income

1100

We can do this for all of the variables we created:

In [5]:
# Show the value of the `rent` variable.
rent

575

In [6]:
# Show the value of the `bills` variable.
bills

150

We use the *name* of each variable to get the *value* from the computer's memory.

Now it is much easier to do the whole calculation, using the names of the variables. 

In [7]:
income - rent - bills

375

Remember Aaron's mean landlord, who wants to increase the rent to £600? 

Now we can easily *re-assign* the variable `rent`. This means that we give it a new value:

In [8]:
# Give `rent` a new value.
rent = 600

Now we can tell the computer to show us the value of the `rent` variable, and we will see the new value:

In [9]:
rent

600

This variable has a new value inside the computer's memory (`600`) - this has changed the old value (`575`).

This means we can do the calculation, with the higher rent, just by using the variable names:

In [10]:
# Calculate again, with the new, bigger `rent` value.
income - rent - bills

350

This is more convenient, and it is harder to make mistakes! If we do not use variables, we have to type the same numbers lots of times. The more we do this, the more errors we will make. For example, maybe we will type 1200 for income, instead of 1100. If we make a `income` variable, we make sure we get the right value from the computer's memory:

In [11]:
# Get the value of `income`, again.
income

1100

Importantly, in assignment statements, the `=` is called the *assignment operator*:

![](https://raw.githubusercontent.com/pxr687/Intro_Comp_Programming/refs/heads/main/images/variable_assignment_all.png)

Variables are very, very, very important in coding. They make our code easy to understand, and they avoid mistakes and errors.

## Exercise 1 - Python, Jupyter and Variables

We will now do our first Python exercise!

*Note*: your assignments, [tests and exams](1_2_Class_Grades_Assignments_Exams) will also be a bit like this.

Click [here 🚀](https://mybinder.org/v2/gh/pxr687/Intro_Comp_Programming/main?urlpath=tree/exercise_1_python_jupyter_intro.ipynb) to launch the exercise.